tests/example_test.go

```diff
@@ -0,0 +1,494 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type parsedExample struct {
+	cloneURL string
+	org      string
+	repo     string
+	path     string
+}
+
+func parseExample(t *testing.T, v string) parsedExample {
+	t.Helper()
+	require.True(t, strings.HasPrefix(v, "https://github.com/"))
+	trimmed := strings.TrimPrefix(v, "https://github.com/")
+	segments := strings.Split(trimmed, "/")
+	require.True(t, len(segments) >= 2)
+
+	var path string
+	if len(segments) > 2 {
+		path = filepath.Join(segments[2:]...)
+	}
+
+	return parsedExample{
+		cloneURL: fmt.Sprintf("https://github.com/%s/%s.git", segments[0], segments[1]),
+		org:      segments[0],
+		repo:     segments[1],
+		path:     path,
+	}
+}
+
+type keyedMutex struct {
+	mutexes sync.Map
+}
+
+func (m *keyedMutex) Lock(key string) func() {
+	value, _ := m.mutexes.LoadOrStore(key, &sync.Mutex{})
+	mutex := value.(*sync.Mutex)
+	mutex.Lock()
+	return func() { mutex.Unlock() }
+}
+
+type stringSet map[string]struct{}
+
+func newStringSet(values ...string) stringSet {
+	s := stringSet{}
+	for _, v := range values {
+		s[v] = struct{}{}
+	}
+	return s
+}
+
+func (ss stringSet) Has(s string) bool {
+	_, ok := ss[s]
+	return ok
+}
+
+func (ss stringSet) Equal(other stringSet) bool {
+	if len(ss) != len(other) {
+		return false
+	}
+	for k := range ss {
+		if !other.Has(k) {
+			return false
+		}
+	}
+	return true
+}
+
+const (
+	csharp     = "c#"
```

Comment:
I don't believe we don't use the literal `c#` as a source language, instead `pulumi convert` accepts `csharp` or `dotnet`

Reply:
Nevermind actually, it was just my terminal that didn't like `c#` but it worked when I used "c#" 😅